sdonboarding.client/src/components/UITemplate/DeleteDataTemplate.jsx

```diff
@@ -0,0 +1,60 @@
+/**
+ * This is a template file for deleting an entity.
+ * Used by Customer, Product, Store, Sale Pages.
+ * 'type' should be passed to this component for it to fetch the required information.
+ */
+
+import { PropTypes } from 'prop-types';
+import { useSendDeleteRequest, useDeleteCancel, useGetDeleteObject } from '../../hooks/DeleteDataHook';
+import { XMarkIcon } from '@heroicons/react/24/solid';
+
+const DeleteDataTemplate = ({ type }) => {
+
+    const deleteObject = useGetDeleteObject(type);
+    const sendDeleteRequest = useSendDeleteRequest(deleteObject, type);
+    const deleteCancel = useDeleteCancel(type);
+
+    //handle Cancel the form
+    const handleCancel = () => {
+        deleteCancel();
+    };
+
+    //handle the form submit event
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+
+        await sendDeleteRequest();
```

Comment:
Is **sendDeleteRequest** a promised based function? What will happen if there's an error there. Should you add error catcher here?